### PR TITLE
[PIP] set osx deploy target for v1.x wheels

### PR DESCRIPTION
**Pull request by szha**
Signed-off-by: Sheng Zha <zhasheng@amazon.com>

## Description ##
set osx deploy target for v1.x wheels

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] set osx deploy target for v1.x wheels

## Comments ##
- This PR includes half of #19110 that hasn't been ported yet.

### CODE REVIEW

File: tools/staticbuild/build_lib_cmake.sh [hunk 1]
```diff
@@ -30,7 +30,12 @@ git submodule update --init --recursive || true
 
 # Build libmxnet.so
 rm -rf build; mkdir build; cd build
-cmake -GNinja -C $cmake_config -DCMAKE_PREFIX_PATH=${DEPS_PATH} -DCMAKE_FIND_ROOT_PATH=${DEPS_PATH} ..
+cmake \
+    -GNinja \
+    -C $cmake_config \
+    -DCMAKE_PREFIX_PATH=${DEPS_PATH} \
+    -DCMAKE_OSX_DEPLOYMENT_TARGET=10.13 \
```
Comment (leezu) [inline tools/staticbuild/build_lib_cmake.sh:37]: Duplicate?

Comment (szha) [reply tools/staticbuild/build_lib_cmake.sh:37]: should have no harm.

Comment (leezu) [reply tools/staticbuild/build_lib_cmake.sh:37]: If there's no benefit, there's no need to add the code. It will just serve to confuse readers

**Comment by szha** [comment]
@mxnet-bot run ci [unix-gpu]

**Pull request merged**